### DISCUSSION TITLE
Seeking Method for Non-Blocking Asynchronous Process in Adonis JS

### DISCUSSION THREAD

**Discussion by KhanAmaan55** [adonisjs/core#4214] [Help] [answered]
Hello,

I'm working with Adonis JS and wondering if there's a method similar to the old Adonis CLI queue or job for running a non-blocking asynchronous process alongside the main controller flow. Any tips or pointers would be appreciated!

Thanks

**Comment by McSneaky** [comment, marked as answer]
It depends on what kind of thing you want. You can use built-in events https://docs.adonisjs.com/guides/events

Or if you can use any non-official queue packages: 
- https://github.com/cavai-research/Adonis-Queue (out of box DB queue, can extend with custom drivers)
- https://github.com/romainlanz/adonis-bull-queue (BullMQ [Redis] based queue)

You can search for community packages from here: https://packages.adonisjs.com/
